src/python/pants/backend/python/dependency_inference/scripts/import_parser.py

```diff
@@ -47,23 +51,23 @@ def _visit_import_stmt(self, node, import_prefix):
         token_iter = tokenize.generate_tokens(lambda: next(node_lines_iter))
 
         def consume_until(string):
-            return list(itertools.takewhile(lambda t: t[1] != string, token_iter))
+            for token in token_iter:
+                if token[1] == string:
+                    return token
```

Comment:
This (and the other modifications that _aren't_ strictly adding to `self.imports`) are fixing a bug I found in the code :raised_hands: